src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -79,9 +86,34 @@ public RuntimeConfig GetConfig()
                 subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
         }
 
+        TrySetupConfigFileWatcher();
         return _runtimeConfig;
     }
 
+    /// <summary>
+    /// Checks if we have already attempted to configure the file watcher, if not
+    /// instantiate the file watcher if we are in the correct scenario. If we
+    /// are not in the correct scenario, do not setup a file watcher but remember
+    /// that we have attempted to do so to avoid repeat checks in future calls.
+    /// Returns true if we instantiate a file watcher.
+    /// </summary>
+    private bool TrySetupConfigFileWatcher()
+    {
+        if (!_hasAttemptedFileWatcherConfiguration)
+        {
+            if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
+            {
+                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
```

Comment:
This is assuming the ConfigLoader HAS to be a FileSystemRuntimeConfigLoader - today it is - but goes against the notion of using a base RuntimeConifgloader. Suggest to put this code in a try catch block to prevent throwing exceptions if the cast fails - adhering to the naming convention of TryXXX